parlai/crowdsourcing/utils/analysis.py

```diff
@@ -185,18 +199,26 @@ def get_task_units(self, task_name: str) -> List[Unit]:
         data_browser = self.get_mephisto_data_browser()
         return data_browser.get_units_for_task_name(task_name)
 
-    def get_units_data(self, task_units: List[Unit]) -> List[dict]:
+    def get_data_from_unit(self, unit: Unit) -> Dict[str, Any]:
+        """
+        Retrieves task data for a single unit.
+        """
+        try:
+            data_browser = self.get_mephisto_data_browser()
```

Comment:
hmm how expensive is the operation of getting the Mephisto data browser? Will doing it once for every unit lead to a slowdown?

Reply:
It is not using Mephisto one directly.  It uses [this](https://github.com/facebookresearch/ParlAI/blob/e062f5d58b88f3e843de8d6635c5aa8018b6a66b/parlai/crowdsourcing/utils/analysis.py#L177) method that keeps the browser object reference, the first time that it accessed it.

Reply:
oh ha great :)